src/pages/lessons/components/UserSettings/UserSettings.tsx

```diff
@@ -8,59 +8,80 @@ import SpeakWordsHelp from "../SpeakWordsHelp";
+  useChangeStenoLayout,
+  useChangeUserSetting,
+  useChangeVoiceUserSetting,
+  useHandleBeatsPerMinute,
+  useHandleDiagramSize,
+  useHandleLimitWordsChange, useHandleRepetitionsChange, useHandleStartFromWordChange, useHandleUpcomingWordsLayout
+} from "./updateUserSetting";
 
 const grabStyle = function () {
   return false;
 };
 
 type Props = {
-  changeShowScoresWhileTyping: (event: any) => void;
-  changeShowStrokesAs: (event: any) => void;
-  changeShowStrokesAsList: (event: any) => void;
-  changeShowStrokesOnMisstroke: (event: any) => void;
-  changeSortOrderUserSetting: (event: any) => void;
-  changeSpacePlacementUserSetting: (event: any) => void;
-  changeStenoLayout: (event: any) => void;
-  changeUserSetting: (event: any) => void;
-  changeVoiceUserSetting: (voiceName: string, voiceURI: string) => void;
   disableUserSettings: boolean;
-  handleBeatsPerMinute: (event: any) => void;
-  handleDiagramSize: (event: any) => void;
-  handleLimitWordsChange: (event: any) => void;
-  handleRepetitionsChange: (event: any) => void;
-  handleStartFromWordChange: (event: any) => void;
-  handleUpcomingWordsLayout: (event: any) => void;
   maxStartFromWord: number;
   revisionMode: boolean;
   totalWordCount: number;
-  userSettings: UserSettingsObjectType;
 };
 
 const UserSettings = ({
-  changeShowScoresWhileTyping,
-  changeShowStrokesAs,
-  changeShowStrokesAsList,
-  changeShowStrokesOnMisstroke,
-  changeSortOrderUserSetting,
-  changeSpacePlacementUserSetting,
-  changeStenoLayout,
-  changeUserSetting,
-  changeVoiceUserSetting,
   disableUserSettings,
-  handleBeatsPerMinute,
-  handleDiagramSize,
-  handleLimitWordsChange,
-  handleRepetitionsChange,
-  handleStartFromWordChange,
-  handleUpcomingWordsLayout,
   maxStartFromWord,
   revisionMode,
   totalWordCount,
-  userSettings,
 }: Props) => {
+  const userSettings = useAtomValue(userSettingsState);
+  const changeShowScoresWhileTyping = useChangeShowScoresWhileTyping()
+  const changeShowStrokesAs = useChangeShowStrokesAs();
+  const changeShowStrokesAsList = useChangeShowStrokesAsList()
+  const changeShowStrokesOnMisstroke = useChangeShowStrokesOnMisstroke();
+  const changeSortOrderUserSetting = useChangeSortOrderUserSetting();
+  const changeSpacePlacementUserSetting = useChangeSpacePlacementUserSetting()
+  const changeStenoLayout = useChangeStenoLayout();
+  const changeUserSetting = useChangeUserSetting();
+  const changeVoiceUserSetting = useChangeVoiceUserSetting();
+  const handleBeatsPerMinute = useHandleBeatsPerMinute();
+  const handleDiagramSize = useHandleDiagramSize();
+  const handleLimitWordsChange = useHandleLimitWordsChange();
+  const handleRepetitionsChange = useHandleRepetitionsChange();
+  const handleStartFromWordChange = useHandleStartFromWordChange();
+  const handleUpcomingWordsLayout = useHandleUpcomingWordsLayout();
   const announceTooltip = useAnnounceTooltip();
 
+  const {setupLesson} = useAppMethods();
+  const mounted = React.useRef(false);
```

Comment:
Let's update the import to `import React, { useEffect, useRef } from "react";` and use `useRef` directly here so we're consistent about our imports.